### PR TITLE
refactor: switch madge circular deps check to ts files

### PR DESCRIPTION
**Pull request by vindard**

### CODE REVIEW

File: core/api/src/app/authentication/login.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ import {
 
 import { upgradeAccountFromDeviceToPhone } from "@/app/accounts"
 import { checkedToEmailCode } from "@/domain/authentication"
-import { isPhoneCodeValid, TwilioClient } from "@/services/twilio"
+import { isPhoneCodeValid, TwilioClient } from "@/services/twilio-service"
```
Comment (vindard) [inline core/api/src/app/authentication/login.ts:38]: Wasn't sure what to do with this one. It seems that the `import twilio from "twilio"` top-line import in that file caused some dependency check issue.

**Review by bodymindarts — APPROVED** [review]

File: core/api/package.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
   "scripts": {
     "tsc-check-noimplicitany": "tsc --noEmit -p tsconfig.no-implicit-any.json",
     "tsc-check": "tsc --noEmit -p tsconfig.d.json && tsc --noEmit",
-    "madge-check": "yarn madge --circular lib/servers/graphql-main-server.js",
+    "madge-check": "yarn madge --circular --extensions ts ./src",
```
Comment (bodymindarts) [inline core/api/package.json:6]: I think we can also delete `yarn` here right?

Comment (bodymindarts) [reply core/api/package.json]: Confirmed - calling `yarn` from within a `package.json` `script` is redundant - by removing it it becomes compatible with `pnpm`.

**Review by bodymindarts — APPROVED** [review]

File: core/api/package.json (hunk 2 above)
Comment (vindard) [reply core/api/package.json]: Yes, I actually did do that in the [other PR](https://github.com/GaloyMoney/galoy/pull/3360/files#diff-96fde044b147bac83eceed2332a32601b6400e185e9034973e0110d25c8175baR6) too

**Pull request merged**